src/features/pilot_management/store/index.ts

```diff
@@ -44,20 +59,16 @@ export class PilotManagementStore extends VuexModule {
     const allPilots = [...payload.map(x => Pilot.Deserialize(x)).filter(x => x)]
     this.Pilots = allPilots
     this.PilotGroups = _.uniq(this.PilotGroups.concat(this.Pilots.map(x => x.Group)).concat(['']))
-    const cmp = function(a, b) {
-      if (a > b) return +1;
-      if (a < b) return -1;
-      return 0;
-    }
-    this.Pilots.sort((a, b) => {
-      return cmp(this.PilotGroups.indexOf(a.Group), this.PilotGroups.indexOf(b.Group)) || cmp(a.SortIndex, b.SortIndex)
-    })
-    this.Pilots.forEach((p, i) => p.SortIndex = i)
+    sortPilotsByGroup(this.Pilots, this.PilotGroups)
+    reindexPilots(this.Pilots)
   }
 
   @Mutation
   private [ADD_PILOT](payload: Pilot): void {
+    payload.SortIndex = this.Pilots.length
```

Comment:
This index isn't permanent; it's simply to position the pilot at the "end" of the `Pilots` array for sorting purposes.